services/ui-src/src/components/modals/AddEditReportModal.test.tsx

```diff
@@ -0,0 +1,63 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import { axe } from "jest-axe";
+//components
+import { AddEditReportModal } from "components";
+import {
+  mockStateUserStore,
+  RouterWrappedComponent,
+} from "utils/testing/setupJest";
+import { useStore } from "utils";
+
+const mockCloseHandler = jest.fn();
+const mockReportHandler = jest.fn();
+
+jest.mock("utils/state/useStore");
+const mockedUseStore = useStore as jest.MockedFunction<typeof useStore>;
+
+const modalComponent = (
+  <RouterWrappedComponent>
+    <AddEditReportModal
+      activeState="AB"
+      reportType={"QM"}
+      modalDisclosure={{
+        isOpen: true,
+        onClose: mockCloseHandler,
+      }}
+      reportHandler={mockReportHandler}
+    />
+  </RouterWrappedComponent>
+);
+
+describe("Test AddEditProgramModal", () => {
+  beforeEach(async () => {
+    mockedUseStore.mockReturnValue(mockStateUserStore);
+    render(modalComponent);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("AddEditReportModal shows the contents", () => {
+    expect(screen.getByText("QMS report name")).toBeTruthy();
+    expect(screen.getByText("Start new")).toBeTruthy();
+  });
+
+  test("AddEditReportModal top close button can be clicked", () => {
+    fireEvent.click(screen.getByText("Close"));
+    expect(mockCloseHandler).toHaveBeenCalledTimes(1);
+  });
+
+  test("AddEditReportModal bottom cancel button can be clicked", () => {
+    fireEvent.click(screen.getByText("Cancel"));
+    expect(mockCloseHandler).toHaveBeenCalledTimes(1);
+  });
```

Comment:
I usually prefer `userEvent` over `fireEvent` &mdash; `fireEvent` is a lower-level library that, well, fires a single event (like "click"). Whereas `userEvent` is higher-level and performs more surrounding checks (like ensuring the element is visible & clickable, and firing "mousedown" and "mouseup" along with "click"). That makes it closer to how users will actually interact with our site, and therefore a higher-quality test.

You will have to `import userEvent from "@testing-library/userEvent"` to make it work though.

```suggestion
  test("AddEditReportModal top close button can be clicked", () => {
    userEvent.click(screen.getByText("Close"));
    expect(mockCloseHandler).toHaveBeenCalledTimes(1);
  });

  test("AddEditReportModal bottom cancel button can be clicked", () => {
    userEvent.click(screen.getByText("Cancel"));
    expect(mockCloseHandler).toHaveBeenCalledTimes(1);
  });
```